### PR TITLE
Use Qt 5.6, msvc2015 and use a copy of the helper scrip for Windows b…

### PR DESCRIPTION
**Pull request by TheOneRing**
…uilds.

As Qt 5.6 no longer supports QtWebkit we continue building Qt 5.5 builds with Webkit.

For builds without Webkit we can probably further reduce the installer size
by removing dll which are no longer needed.

### CODE REVIEW

**Comment by TheOneRing** [comment]
It might make sense to drop 5.5 and thus the webkit preview.

**Comment by TheOneRing** [comment]
Before we can merge this I need to update the new msvc runtime

**Comment by digitalcircuit** [comment]
Ack, I apologize for duplicating work via #205.  Your pull request is a much more complete approach!

You'll need to drop Visual Studio 2013 support to account for recent code changes, which unfortunately means no 64-bit builds with web previews thanks to AppVeyor lacking Qt 5.5 for ```msvc2013```.

Qt 5.5 with QtWebKit should still be viable via ```mingw492_32```.

(And very minor, you might want to shorten the first commit line to fit within 52 characters)

File: appveyor.yml [hunk 1]
```diff
@@ -116,19 +116,25 @@ build_script:
...
 
     matrix:
     #msvc
-    - COMPILER: msvc2013_64
-    - COMPILER: msvc2013
+    - COMPILER: msvc2015_64
+      QT_VER: 5.6
+    - COMPILER: msvc2015
+      QT_VER: 5.6
+
```
Comment (AlD) [inline appveyor.yml]: Whitespace error

File: appveyor.yml [hunk 2]
```diff
@@ -116,19 +116,25 @@ build_script:
...
+      QT_VER: 5.6
+    - COMPILER: msvc2015
+      QT_VER: 5.6
+  
     #mingw
+    - COMPILER: mingw49_32
+      QT_VER: 5.6
     - COMPILER: mingw492_32
+      QT_VER: 5.5
+
```
Comment (AlD) [inline appveyor.yml]: Whitespace error

File: scripts/build/appveyorHelp.psm1 [hunk 3]
```diff
@@ -0,0 +1,368 @@
...
+$env:PATH="$env:PATH;$script:INSTALL_DIR"
+
+if(!$env:CI -eq "true")
+{
+    function Push-AppveyorArtifact()
+    {
+        Write-Host "Push-AppveyorArtifact $ARGS"
+    }
+    
+    function Start-FileDownload([string] $url, [string] $out)
```
Comment (AlD) [inline scripts/build/appveyorHelp.psm1]: Whitespace error

File: scripts/build/appveyorHelp.psm1 [hunk 4]
```diff
@@ -0,0 +1,368 @@
...
+$CMAKE_INSTALL_ROOT="`"$INSTALL_DIR`"" -replace "\\", "/"
+$env:PATH="$env:PATH;$script:INSTALL_DIR"
+
+if(!$env:CI -eq "true")
+{
+    function Push-AppveyorArtifact()
+    {
+        Write-Host "Push-AppveyorArtifact $ARGS"
+    }
+
```
Comment (AlD) [inline scripts/build/appveyorHelp.psm1]: Whitespace error

File: scripts/build/appveyorHelp.psm1 [hunk 5]
```diff
@@ -0,0 +1,368 @@
...
+    {
+        if(!$out)
+        {
+            $out = $url.SubString($url.LastIndexOf("/"))
+        }
+        wget $url -Outfile $out
+    }
+}
+
+
```
Comment (AlD) [inline scripts/build/appveyorHelp.psm1]: Whitespace error

File: scripts/build/appveyorHelp.psm1 [hunk 6]
```diff
@@ -0,0 +1,368 @@
...
+function Get-QtDir()
+{
+    $ver = 5.5
+    if($env:QT_VER)
+    {
+        $ver = $env:QT_VER
+    }
+    return "C:\Qt\$ver\$env:COMPILER\"
+}
```
Comment (AlD) [inline scripts/build/appveyorHelp.psm1]: Whitespace error

File: scripts/build/appveyorHelp.psm1 [hunk 7]
```diff
@@ -0,0 +1,368 @@
...
+    }
+    return "C:\Qt\$ver\$env:COMPILER\"
+} 
+
+function SETUP-QT()
+{
+    [string] $compiler=$env:COMPILER
+    $qtDir = Get-QtDir
+    $script:QT_BINARY_DIRS = @($qtDir)
+
```
Comment (AlD) [inline scripts/build/appveyorHelp.psm1]: Whitespace error

File: scripts/build/appveyorHelp.psm1 [hunk 8]
```diff
@@ -0,0 +1,368 @@
...
+                "msvc2015" = "VS140COMNTOOLS"
+            }
+
+        $compilerVar = $compilerDirs[$compiler.Split("_")[0]]
+        $compilerDir = (get-item -path "env:\$($compilerVar)").Value
+        BAT-CALL "$compilerDir\..\..\VC\vcvarsall.bat" $arch
+        $script:MAKE="nmake"
+        $script:CMAKE_GENERATOR="NMake Makefiles"
+        if($arch -eq "x86")
+        {
```
Comment (AlD) [inline scripts/build/appveyorHelp.psm1]: Whitespace error

File: scripts/build/appveyorHelp.psm1 [hunk 9]
```diff
@@ -0,0 +1,368 @@
...
+
+function Init([string[]] $chocoDeps, [System.Collections.Specialized.OrderedDictionary] $cmakeModules)
+{
+    $script:MAKE=""
+    $script:CMAKE_GENERATOR=""
+    $script:STRIP=$null
+
+    mkdir -Force $env:APPVEYOR_BUILD_FOLDER\work\image | Out-Null
+    mkdir -Force $env:APPVEYOR_BUILD_FOLDER\work\build | Out-Null
+
```
Comment (AlD) [inline scripts/build/appveyorHelp.psm1]: Whitespace error

File: scripts/build/appveyorHelp.psm1 [hunk 10]
```diff
@@ -0,0 +1,368 @@
...
+{
+    $script:MAKE=""
+    $script:CMAKE_GENERATOR=""
+    $script:STRIP=$null
+
+    mkdir -Force $env:APPVEYOR_BUILD_FOLDER\work\image | Out-Null
+    mkdir -Force $env:APPVEYOR_BUILD_FOLDER\work\build | Out-Null
+    
+    SETUP-QT
+
```
Comment (AlD) [inline scripts/build/appveyorHelp.psm1]: Whitespace error

File: scripts/build/appveyorHelp.psm1 [hunk 11]
```diff
@@ -0,0 +1,368 @@
...
+    mkdir -Force $env:APPVEYOR_BUILD_FOLDER\work\image | Out-Null
+    mkdir -Force $env:APPVEYOR_BUILD_FOLDER\work\build | Out-Null
+    
+    SETUP-QT
+    
+    if($chocoDeps -contains "ninja") {
+        $script:CMAKE_GENERATOR="Ninja"
+        $script:MAKE="ninja"
+    }
+
```
Comment (AlD) [inline scripts/build/appveyorHelp.psm1]: Whitespace error

**Comment by AlD** [comment]
Thanks a bunch!

**Pull request merged**